spec/numberformat.html

```diff
@@ -1251,5 +1506,313 @@ <h1>ComputeExponentForMagnitude ( _numberFormat_, _magnitude_ )</h1>
+        1. Let _d2_ be <emu-eqn>_r2_ – _x_</emu-eqn>.
+        1. If _d1_ &lt; _d2_, return _r1_.
+        1. If _d2_ &lt; _d1_, return _r2_.
+        1. Assert: _d1_ is equal to _d2_.
+        1. If _unsignedRoundingMode_ is ~half-zero~, return _r1_.
+        1. If _unsignedRoundingMode_ is ~half-infinity~, return _r2_.
+        1. Assert: _unsignedRoundingMode_ is ~half-even~.
+        1. Let _cardinality_ be <emu-eqn>(_r1_ / (_r2_ – _r1_)) modulo 2</emu-eqn>.
+        1. If _cardinality_ is 0, return _r1_.
+        1. Return _r2_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-partitionnumberrangepattern" aoid="PartitionNumberRangePattern">
+      <h1>PartitionNumberRangePattern ( _numberFormat_, _x_, _y_ )</h1>
+      <p>
+        The abstract operation PartitionNumberRangePattern creates the parts for a localized number range according to _x_ (which must be an Intl mathematical value), _y_ (which must be an Intl mathematical value), and the formatting options of _numberFormat_ (which must be an object initialized as NumberFormat). The following steps are taken:
+      </p>
+      <emu-alg>
+        1. If _x_ is ~not-a-number~ or _y_ is ~not-a-number~, throw a *RangeError* exception.
+        1. Let _result_ be a new empty List.
+        1. Let _xResult_ be ? PartitionNumberPattern(_numberFormat_, _x_).
+        1. Let _yResult_ be ? PartitionNumberPattern(_numberFormat_, _y_).
+        1. If _xResult_ is equal to _yResult_, then
+          1. Let _appxResult_ be ? FormatApproximately(_numberFormat_, _xResult_).
+          1. For each element _r_ in _appxResult_, do
+            1. Set _r_.[[Source]] to *"shared"*.
+          1. Return _appxResult_.
+        1. For each element _r_ in _xResult_, do
+          1. Set _r_.[[Source]] to *"startRange"*.
+        1. Add all elements in _xResult_ to _result_ in order.
+        1. Let _rangeSeparator_ be an ILND String value used to separate two numbers.
+        1. Append a new Record { [[Type]]: *"literal"*, [[Value]]: _rangeSeparator_, [[Source]]: *"shared"* } element to _result_.
+        1. For each element _r_ in _yResult_, do
+          1. Set _r_.[[Source]] to *"endRange"*.
+        1. Add all elements in _yResult_ to _result_ in order.
+        1. Return ! CollapseNumberRange(_result_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-formatapproximately" aoid="FormatApproximately">
+      <h1>FormatApproximately ( _numberFormat_, _result_ )</h1>
+      <p>
+        The FormatApproximately abstract operation modifies _result_, which must be a List of Record values as described in PartitionNumberPattern, by adding a new Record for the approximately sign, which may depend on _numberFormat_ (which must be an object initialized as NumberFormat). The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _i_ be an index into _result_, determined by an implementation-defined algorithm based on _numberFormat_ and _result_.
```

Comment:
nit: could we add a bit more info to guide implementations here at all?

Reply:
Added a sentence